### ISSUE TITLE
False positive from no-implicit-conversion

### ISSUE THREAD

**Issue by glenjamin** [eslint/eslint#6199]
**What version of ESLint are you using?**
v2.10.2

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
`{"no-implicit-coercion":"error"}`

**What did you do? Please include the actual source code causing the issue.**
```js
var x = y * 1/4
```
```sh
eslint temp.js --rule '{"no-implicit-coercion":"error"}'
```

**What did you expect to happen?**

No error

**What actually happened? Please include the actual, raw output from ESLint.**

```
  1:9  error  use `Number(y)` instead  no-implicit-coercion
```

**Comment by ilyavolodin** [comment]
That looks like a bug. Don't know why it would think this is coercion, since we have no idea what the values of either `x` or `y` are. @mysticatea could you take a look please?

**Comment by glenjamin** [comment]
I'm pretty sure the reason is that `y * 1` is a coercion, and it doesn't attempt to apply operator precedence rules so never sees the `/4`.

In this case I changed the line to `y / 4` to avoid the problem, so it's not hugely problematic.

**Comment by ilyavolodin** [comment]
Ah, that makes perfect sense. But I don't know how we fix the rule without implementing some very complex logic. Maybe if there's more then one operator in the expression, we don't consider it coercion?

**Comment by michaelficarra** [comment]
`y * 1/4`, despite the deceptive spacing, is disambiguated as `(y * 1) / 4`. This rule is working as intended, reporting your usage of `y * 1`, which is useless other than for type coercion.

**Comment by glenjamin** [comment]
Oh whoops, for some reason I thought divide was higher than multiply - but I was mistaken!
